README.md

```diff
@@ -1,2 +1,981 @@
+                 U nanosleep
+0000000000000000 T pal_sleep
+```
+
+As expected, _pal_thread_pthread.o_ exports `pal_sleep` and request an implementation of `nanosleep`, that shall be provided by the pthread library. 
+
+#### Other dependencies
+
+The system may have other dependencies, for example the way that it halt in case of a critical error. It is better to keep it in the PAL layer, even if the current implementation work for all platforms, it will facilitate change if a new platform requires a different implementation in the future. Better safe than sorry.
+
+### Creating the PAL libraries
+
+Up to now, we build our `.c` files directly using _gcc_ or _cl_, and we didn't link it to create a library or a executable. Now, let's move to cmake and create a project that will compile the `.c` files and link the object files creating a static library. Cmake uses _CMakeLists.txt_ to define the project. The PAL _CMakeLists.txt_ shall create 3 libraries, one for each board, and it shall chose the right thread library to compile and link, depending on the OS. 
+
+Cmake is a very powerful tool, for this sample, we will use only the ability to set variables, create projects, and do some glue logic.
+
+#### Defining libraries using _config.cmake_
+
+The set of `.c` files that will compose each library shall be enough to cover all APIs exposed by the header _pal.h_, in our case, it needs one adapter for the temperature sensor, fan, led, exception, and thread. The definition of these configurations are in the _config.cmake_. A _.cmake_ file is a file that can be included in any _CMakeLists.txt_ to export variables or functions, in this sample, the _config.cmake_ expose the list of configurations and the functions to create samples and tests. 
+
+Observe that we defined all possible configurations in the _config.cmake_, independent of the OS. As a convention, we define configurations pal_win_x for windows, and pal_linux_x for windows 
+
+```cmake
+##
+# Description of PAL configurations for Windows.
+##
+list(APPEND pal_win_1
+    pal_version.c
+    pal_temperature_a.c
+    pal_led.c
+    pal_fan_a.c
+    pal_exception.c
+    pal_thread_windows.c
+)
+
+list(APPEND pal_win_2
+    pal_version.c
+    pal_temperature_b.c
+    pal_led.c
+    pal_fan_b.c
+    pal_exception.c
+    pal_thread_windows.c
+)
+
+list(APPEND pal_win_3
+    pal_version.c
+    pal_temperature_b.c
+    pal_led.c
+    pal_fan_c.c
+    pal_exception.c
+    pal_thread_windows.c
+)
+
+##
+# Description of PAL configurations for Linux.
+##
+list(APPEND pal_linux_1
+    pal_version.c
+    pal_temperature_a.c
+    pal_led.c
+    pal_fan_a.c
+    pal_exception.c
+    pal_thread_pthread.c
+)
+
+list(APPEND pal_linux_2
+    pal_version.c
+    pal_temperature_b.c
+    pal_led.c
+    pal_fan_b.c
+    pal_exception.c
+    pal_thread_pthread.c
+)
+
+list(APPEND pal_linux_3
+    pal_version.c
+    pal_temperature_b.c
+    pal_led.c
+    pal_fan_c.c
+    pal_exception.c
+    pal_thread_pthread.c
+)
+```
```

Comment:
nit: these will likely get out of sync with implementations. maybe just add a link to it or a short example?